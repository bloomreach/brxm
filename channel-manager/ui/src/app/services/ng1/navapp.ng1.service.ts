/*!
 * Copyright 2021 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { InjectionToken } from '@angular/core';
import { ClientError, NavLocation, ParentConfig, UserSettings } from '@bloomreach/navapp-communication';

export interface Ng1NavappService {
  subscribe: (api: string, callback: () => Promise<any>) => void;
  unsubscribe: (api: string, callback: () => Promise<any>) => void;
  connect: () => Promise<void>;
  getConfig(): Promise<ParentConfig>;
  getUserSettings(): Promise<UserSettings>;
  updateNavLocation(location: NavLocation): Promise<void>;
  navigate(location: NavLocation): Promise<void>;
  showMask(): Promise<void>;
  hideMask(): Promise<void>;
  showBusyIndicator(): Promise<void>;
  hideBusyIndicator(): Promise<void>;
  onUserActivity(): Promise<void>;
  onSessionExpired(): Promise<void>;
  onError(error: ClientError): Promise<void>;
}

export const NG1_NAVAPP_SERVICE = new InjectionToken<Ng1NavappService>('NG1_NAVAPP_SERVICE');
