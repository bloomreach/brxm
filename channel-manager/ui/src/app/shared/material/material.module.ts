/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { NgModule } from '@angular/core';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule, MatIconRegistry } from '@angular/material/icon';
import { MatListModule } from '@angular/material/list';
import { DomSanitizer } from '@angular/platform-browser';

import { registerSvgIcons } from './register-svg-icons';

@NgModule({
  exports: [
    MatButtonModule,
    MatListModule,
    MatIconModule,
  ],
})
export class MaterialModule {
  constructor(private readonly iconRegistry: MatIconRegistry, private readonly domSanitizer: DomSanitizer) {
    this.registerCustomIcons();
  }

  private registerCustomIcons(): void {
    registerSvgIcons(this.iconRegistry, this.domSanitizer);
  }
}
