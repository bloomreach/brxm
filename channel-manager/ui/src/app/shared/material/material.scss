/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

@import '~@angular/material/theming';
@import '~@bloomreach/frontend-theme/palettes';
@import 'variables';

$custom-typography: mat-typography-config(
  $font-family: inspect($font-family)
);

@include mat-core($custom-typography);

$primary: mat-palette($blue);
$accent: mat-palette($orange);
$warn: mat-palette($red);
$theme: mat-light-theme($primary, $accent, $warn);

@include angular-material-theme($theme);

// default body text color
.mat-typography {
  color: map-get($text-color, "dark");
}
