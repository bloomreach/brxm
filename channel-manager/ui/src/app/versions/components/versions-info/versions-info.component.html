<!--
  Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
  -->
<div class="mat-typography">
  <mat-selection-list class="qa-version-list"
                      *ngIf="versionsInfo.versions.length === 0">
    <h3 matSubheader
        class="qa-version-list-header">
      {{ 'NO_VERSIONS_AVAILABLE' | translate }}
    </h3>
    <mat-divider></mat-divider>
  </mat-selection-list>

  <mat-selection-list *ngIf="versionsInfo.versions.length > 0"
                      color="primary"
                      class="qa-version-list"
                      [multiple]="false"
                      (selectionChange)="selectVersion($event.option.value)">
    <h3 matSubheader
        class="qa-version-list-header">
      {{ 'VERSIONS_LIST_HEADER' | translate }}

      <mat-progress-bar *ngIf="actionInProgress"
                        mode="indeterminate">
      </mat-progress-bar>
    </h3>

    <mat-divider></mat-divider>

    <mat-list-option *ngFor="let version of versionsInfo.versions; first as isFirst"
                     [class]="'qa-version-' + version.jcrUUID"
                     [value]="version.jcrUUID"
                     [selected]="unpublishedVariantId === version.jcrUUID">
      <p matLine
         *ngIf="isFirst"
         class="qa-latest-version">
        {{ 'LATEST_VERSION' | translate }}
      </p>

      <p matLine
         *ngIf="!isFirst"
         class="qa-version-timestamp">
        {{ version.timestamp | date:'MMM d, y hh:mm' }}
      </p>

      <p matLine
         *ngIf="!isFirst"
         class="qa-version-username">
        {{ version.userName || ('SYSTEM' | translate) }}
      </p>

      <div matLine
           class="version-actions">
        <button *ngIf="unpublishedVariantId === version.jcrUUID && isFirst"
                mat-stroked-button
                color="primary"
                [disabled]="actionInProgress"
                class="qa-new-version-action"
                (click)="createVersion()">
          <mat-icon svgIcon="plus"></mat-icon>
          {{ 'VERSION' | translate }}
        </button>

        <button *ngIf="unpublishedVariantId === version.jcrUUID && !isFirst"
                mat-stroked-button
                color="primary"
                [disabled]="actionInProgress"
                class="qa-restore-version-action"
                (click)="restoreVersion(version.jcrUUID)">
          {{ 'RESTORE' | translate }}
        </button>
      </div>
      <mat-divider></mat-divider>
    </mat-list-option>
  </mat-selection-list>
</div>
