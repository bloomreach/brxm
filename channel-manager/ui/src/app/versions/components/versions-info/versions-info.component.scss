/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */


.mat-line {
  font-size: 14px !important;
}

.mat-line.version-actions {
  display: none;
}

.mat-list-single-selected-option {
  background-color: #e7f1f9;

  .mat-line:first-child {
    font-weight: bold !important;
  }

  .version-actions {
    display: flex;
    justify-content: flex-end;

    button:not(:first-child) {
      margin-left: 8px;
    }
  }
}
