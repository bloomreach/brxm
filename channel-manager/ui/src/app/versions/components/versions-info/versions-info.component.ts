/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { ChangeDetectorRef, Component, Inject, Input, OnInit } from '@angular/core';

import { Ng1ChannelService, NG1_CHANNEL_SERVICE } from '../../../services/ng1/channel.ng1service';
import { Ng1ContentService, NG1_CONTENT_SERVICE } from '../../../services/ng1/content.ng1.service';
import { Ng1IframeService, NG1_IFRAME_SERVICE } from '../../../services/ng1/iframe.ng1service';
import { Ng1WorkflowService, NG1_WORKFLOW_SERVICE } from '../../../services/ng1/workflow.ng1.service';
import { VersionsInfo } from '../../models/versions-info.model';

@Component({
  selector: 'em-versions-info',
  templateUrl: './versions-info.component.html',
  styleUrls: ['./versions-info.component.scss'],
})
export class VersionsInfoComponent implements OnInit {
  @Input()
  documentId!: string;

  @Input()
  branchId!: string;

  @Input()
  unpublishedVariantId!: string;

  versionsInfo?: VersionsInfo;

  actionInProgress = false;

  constructor(
    @Inject(NG1_CONTENT_SERVICE) private readonly ng1ContentService: Ng1ContentService,
    @Inject(NG1_IFRAME_SERVICE) private readonly ng1IframeService: Ng1IframeService,
    @Inject(NG1_CHANNEL_SERVICE) private readonly ng1ChannelService: Ng1ChannelService,
    @Inject(NG1_WORKFLOW_SERVICE) private readonly ng1WorkflowService: Ng1WorkflowService,
    private readonly changeDetector: ChangeDetectorRef,
  ) { }

  ngOnInit(): void {
    setTimeout(() => {
      // Angular Element inputs are not yet initialized onInit
      this.getVersionsInfo();
    });
  }

  async getVersionsInfo(): Promise<void> {
    this.actionInProgress = true;
    this.versionsInfo = await this.ng1ContentService.getDocumentVersionsInfo(this.documentId, this.branchId);

    this.changeDetector.markForCheck();
    this.actionInProgress = false;
  }

  async selectVersion(versionUUID: string): Promise<void> {
    this.actionInProgress = true;
    const newPath = this.createVersionPath(versionUUID);
    await this.ng1IframeService.load(newPath);
    this.actionInProgress = false;
  }

  async createVersion(): Promise<void> {
    this.actionInProgress = true;
    await this.ng1WorkflowService.createWorkflowAction(this.documentId, 'version');
    await this.getVersionsInfo();
  }

  async restoreVersion(versionUUID: string): Promise<void> {
    this.actionInProgress = true;
    await this.ng1WorkflowService.createWorkflowAction(this.documentId, 'restore', versionUUID);
    const currentPath = this.ng1IframeService.getCurrentRenderPathInfo();
    const renderPath = this.ng1ChannelService.makeRenderPath(currentPath);
    await this.ng1IframeService.load(renderPath);
    await this.getVersionsInfo();
  }

  private createVersionPath(selectedVersionUUID: string): string {
    const currentPath = this.ng1IframeService.getCurrentRenderPathInfo();
    const versionParam = `br_version_uuid=${selectedVersionUUID}`;
    const index = this.versionsInfo?.versions.findIndex(v => v.jcrUUID === selectedVersionUUID);
    const homePageRenderPath = this.ng1ChannelService.getHomePageRenderPathInfo();
    const renderPath = this.ng1ChannelService.makeRenderPath(currentPath.replace(homePageRenderPath, ''));

    if (index && index <= 0) {
      return renderPath;
    }

    if (renderPath.includes('?')) {
      return `${renderPath}&${versionParam}`;
    }

    return `${renderPath}?${versionParam}`;
  }

}
