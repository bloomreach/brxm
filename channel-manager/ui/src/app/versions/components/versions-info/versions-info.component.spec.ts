/*!
 * Copyright 2020 Bloomreach. All rights reserved. (https://www.bloomreach.com/)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *  http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
import { MatIconModule } from '@angular/material/icon';
import { MatIconTestingModule } from '@angular/material/icon/testing';
import { MatListModule } from '@angular/material/list';
import { MatProgressBarModule } from '@angular/material/progress-bar';
import { TranslateModule } from '@ngx-translate/core';

import { Ng1ChannelService, NG1_CHANNEL_SERVICE } from '../../../services/ng1/channel.ng1service';
import { NG1_CONTENT_SERVICE } from '../../../services/ng1/content.ng1.service';
import { Ng1IframeService, NG1_IFRAME_SERVICE } from '../../../services/ng1/iframe.ng1service';
import { Ng1WorkflowService, NG1_WORKFLOW_SERVICE } from '../../../services/ng1/workflow.ng1.service';
import { VersionsInfo } from '../../models/versions-info.model';

import { VersionsInfoComponent } from './versions-info.component';

describe('VersionsInfoComponent', () => {
  let component: VersionsInfoComponent;
  let componentEl: HTMLElement;
  let fixture: ComponentFixture<VersionsInfoComponent>;
  let ng1IframeService: Ng1IframeService;
  let ng1ChannelService: Ng1ChannelService;
  let ng1WorkflowService: Ng1WorkflowService;

  const date = Date.parse('11/08/2020 16:03');
  const path = '/some/test/path';
  const renderPath = `${path}?withParam=test`;
  const homePageRenderPath = '';
  const firstVersionUUID = 'testId';
  const secondVersionUUID = 'testVariantId';
  const mockVersionsInfo = {
    versions: [
      {
        jcrUUID: firstVersionUUID,
        userName: 'testUserName',
        timestamp: date,
      },
      {
        jcrUUID: secondVersionUUID,
        userName: 'testUserName2',
        timestamp: date,
      },
    ],
    restoreEnabled: true,
  } as VersionsInfo;

  beforeEach(() => {
    const contentServiceMock = {
      getDocumentVersionsInfo: jest.fn(() => Promise.resolve(mockVersionsInfo)),
    };

    const channelServiceMock = {
      makeRenderPath: () => path,
      getHomePageRenderPathInfo: () => homePageRenderPath,
    };

    const iframeServiceMock = {
      getCurrentRenderPathInfo: () => path,
      load: jest.fn(() => Promise.resolve()),
    };

    const workflowServiceMock = {
      createWorkflowAction: jest.fn(() => { }),
    };

    TestBed.configureTestingModule({
      declarations: [VersionsInfoComponent],
      imports: [
        MatListModule,
        MatIconModule,
        MatIconTestingModule,
        MatProgressBarModule,
        TranslateModule.forRoot(),
      ],
      providers: [
        { provide: NG1_CONTENT_SERVICE, useValue: contentServiceMock },
        { provide: NG1_CHANNEL_SERVICE, useValue: channelServiceMock },
        { provide: NG1_IFRAME_SERVICE, useValue: iframeServiceMock },
        { provide: NG1_WORKFLOW_SERVICE, useValue: workflowServiceMock },
      ],
    });

    ng1IframeService = TestBed.inject(NG1_IFRAME_SERVICE);
    ng1ChannelService = TestBed.inject(NG1_CHANNEL_SERVICE);
    ng1WorkflowService = TestBed.inject(NG1_WORKFLOW_SERVICE);
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(VersionsInfoComponent);
    component = fixture.componentInstance;
    componentEl = fixture.nativeElement;

    component.documentId = 'testDocumentId';
    component.branchId = 'projectId';
    component.unpublishedVariantId = 'testId';
    fixture.detectChanges();
  });

  describe('showing versions', () => {
    it('should show header indicating no versions are available', () => {
      expect(componentEl).toMatchSnapshot();
    });

    it('should show list of versions', fakeAsync(() => {
      component.ngOnInit();
      tick();
      fixture.detectChanges();

      expect(componentEl).toMatchSnapshot();
    }));
  });

  describe('selecting version', () => {
    beforeEach(fakeAsync(() => {
      component.ngOnInit();
      tick();
      fixture.detectChanges();
    }));

    it('should add the version param to the url and load that url', () => {
      jest.spyOn(ng1IframeService, 'load');

      const versionItem = componentEl.querySelector<HTMLElement>(`.qa-version-${secondVersionUUID}`);
      versionItem?.click();

      expect(ng1IframeService.load).toHaveBeenCalledWith(`${path}?br_version_uuid=${secondVersionUUID}`);
    });

    it('should append the version param to the url if params are already present and load that url', () => {
      jest.spyOn(ng1IframeService, 'load');
      jest.spyOn(ng1ChannelService, 'makeRenderPath').mockReturnValueOnce(renderPath);

      const versionItem = componentEl.querySelector<HTMLElement>(`.qa-version-${secondVersionUUID}`);
      versionItem?.click();

      expect(ng1IframeService.load).toHaveBeenCalledWith(`${renderPath}&br_version_uuid=${secondVersionUUID}`);
    });

    it('should show the selected version', () => {
      jest.spyOn(ng1IframeService, 'load').mockImplementationOnce(() => {
        component.unpublishedVariantId = secondVersionUUID;
        return Promise.resolve();
      });

      const versionItem = componentEl.querySelector<HTMLElement>(`.qa-version-${secondVersionUUID}`);
      versionItem?.click();
      fixture.detectChanges();

      const versionList = componentEl.querySelector<HTMLElement>('.qa-version-list');
      expect(versionList).toMatchSnapshot();
    });
  });

  describe('restoring a version', () => {
    beforeEach(fakeAsync(() => {
      component.ngOnInit();
      tick();
      fixture.detectChanges();
    }));

    it('should show restore button for other versions when selected', async () => {
      jest.spyOn(ng1IframeService, 'load').mockImplementationOnce(() => {
        component.unpublishedVariantId = secondVersionUUID;
        return Promise.resolve();
      });

      await component.selectVersion(secondVersionUUID);
      fixture.detectChanges();

      expect(componentEl).toMatchSnapshot();
    });

    it('should not show restore button for first version when selected', async () => {
      jest.spyOn(ng1IframeService, 'load').mockImplementationOnce(() => {
        component.unpublishedVariantId =  firstVersionUUID;
        return Promise.resolve();
      });

      await component.selectVersion(firstVersionUUID);
      fixture.detectChanges();

      expect(componentEl).toMatchSnapshot();
    });

    it('should call to restore', async () => {
      jest.spyOn(ng1IframeService, 'load').mockImplementationOnce(() => {
        component.unpublishedVariantId = secondVersionUUID;
        return Promise.resolve();
      });

      await component.selectVersion(secondVersionUUID);
      fixture.detectChanges();

      const restoreButton = componentEl.querySelector<HTMLButtonElement>('.qa-restore-version-action');
      restoreButton?.click();

      expect(ng1WorkflowService.createWorkflowAction).toHaveBeenCalledWith(component.documentId, 'restore', secondVersionUUID);
    });
  });

  describe('create version', () => {
    it('should show version button for first version when selected', async () => {
      jest.spyOn(ng1IframeService, 'load').mockImplementationOnce(() => {
        component.unpublishedVariantId = firstVersionUUID;
        return Promise.resolve();
      });

      await component.selectVersion(firstVersionUUID);
      fixture.detectChanges();

      expect(componentEl).toMatchSnapshot();
    });

    it('should not show version button for other versions when selected', async () => {
      jest.spyOn(ng1IframeService, 'load').mockImplementationOnce(() => {
        component.unpublishedVariantId = secondVersionUUID;
        return Promise.resolve();
      });

      await component.selectVersion(secondVersionUUID);
      fixture.detectChanges();

      expect(componentEl).toMatchSnapshot();
    });

    it('should call to create version', fakeAsync(() => {
      component.ngOnInit();
      tick();
      fixture.detectChanges();

      const newVersionButton = componentEl.querySelector<HTMLButtonElement>(`.qa-new-version-action`);
      newVersionButton?.click();
      fixture.detectChanges();

      expect(ng1WorkflowService.createWorkflowAction).toHaveBeenCalledWith(component.documentId, 'version');
    }));
  });

  describe('progress indicator', () => {
    it('should show when action is being performed that takes a while to complete', fakeAsync(() => {
      component.ngOnInit();
      tick();
      fixture.detectChanges();

      component.restoreVersion(secondVersionUUID);
      fixture.detectChanges();

      const header = componentEl.querySelector<HTMLElement>('.qa-version-list-header');
      expect(header).toMatchSnapshot();
    }));
  });

  describe('onDestroy', () => {
    it.only('should select the latest version when removing the component from the dom', fakeAsync(() => {
      jest.spyOn(component, 'selectVersion');

      component.ngOnInit();
      tick();
      component.unpublishedVariantId = secondVersionUUID;
      component.ngOnDestroy();

      expect(component.selectVersion).toHaveBeenCalledWith(firstVersionUUID);
    }));
  });
});
