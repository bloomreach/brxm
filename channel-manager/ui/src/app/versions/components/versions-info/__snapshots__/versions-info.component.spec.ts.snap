// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VersionsInfoComponent create version should not show version button for other versions when selected 1`] = `
<div
  id="root9"
>
  <div
    class="mat-typography"
  >
    
    <mat-selection-list
      aria-disabled="false"
      aria-multiselectable="true"
      class="qa-version-list mat-selection-list mat-list-base"
      role="listbox"
      tabindex="-1"
    >
      <h3
        class="qa-version-list-header mat-subheader"
        matsubheader=""
      >
         NO_VERSIONS_AVAILABLE 
      </h3>
      <mat-divider
        aria-orientation="horizontal"
        class="mat-divider mat-divider-horizontal"
        role="separator"
      />
    </mat-selection-list>
    
  </div>
</div>
`;

exports[`VersionsInfoComponent create version should show version button for first version when selected 1`] = `
<div
  id="root8"
>
  <div
    class="mat-typography"
  >
    
    <mat-selection-list
      aria-disabled="false"
      aria-multiselectable="true"
      class="qa-version-list mat-selection-list mat-list-base"
      role="listbox"
      tabindex="-1"
    >
      <h3
        class="qa-version-list-header mat-subheader"
        matsubheader=""
      >
         NO_VERSIONS_AVAILABLE 
      </h3>
      <mat-divider
        aria-orientation="horizontal"
        class="mat-divider mat-divider-horizontal"
        role="separator"
      />
    </mat-selection-list>
    
  </div>
</div>
`;

exports[`VersionsInfoComponent restoring a version should not show restore button for first version when selected 1`] = `
<div
  id="root6"
>
  <div
    class="mat-typography"
  >
    
    
    <mat-selection-list
      aria-disabled="false"
      aria-multiselectable="false"
      class="qa-version-list mat-selection-list mat-list-base"
      color="primary"
      role="listbox"
      tabindex="0"
    >
      <h3
        class="qa-version-list-header mat-subheader"
        matsubheader=""
      >
         VERSIONS_LIST_HEADER 
      </h3>
      <mat-divider
        aria-orientation="horizontal"
        class="mat-divider mat-divider-horizontal"
        role="separator"
      />
      
      <mat-list-option
        aria-disabled="false"
        aria-selected="true"
        class="qa-version-testId mat-primary mat-list-single-selected-option"
        role="option"
        tabindex="-1"
      >
        <div
          class="mat-list-item-content mat-list-item-content-reverse"
        >
          <div
            class="mat-list-item-ripple mat-ripple"
            mat-ripple=""
          />
          
          <div
            class="mat-list-text"
          >
            <p
              class="qa-version-timestamp mat-line"
              matline=""
            >
               Nov 8, 2020 04:03 
            </p>
            <p
              class="qa-version-username mat-line"
              matline=""
            >
               testUserName 
            </p>
            <div
              class="version-actions mat-line"
              matline=""
            >
              
              <button
                class="qa-new-version-action"
                mat-stroked-button=""
              >
                 VERSION 
              </button>
              
            </div>
            <mat-divider
              aria-orientation="horizontal"
              class="mat-divider mat-divider-horizontal"
              role="separator"
            />
          </div>
        </div>
      </mat-list-option>
      <mat-list-option
        aria-disabled="false"
        aria-selected="false"
        class="qa-version-testVariantId mat-primary"
        role="option"
        tabindex="-1"
      >
        <div
          class="mat-list-item-content mat-list-item-content-reverse"
        >
          <div
            class="mat-list-item-ripple mat-ripple"
            mat-ripple=""
          />
          
          <div
            class="mat-list-text"
          >
            <p
              class="qa-version-timestamp mat-line"
              matline=""
            >
               Nov 8, 2020 04:03 
            </p>
            <p
              class="qa-version-username mat-line"
              matline=""
            >
               testUserName2 
            </p>
            <div
              class="version-actions mat-line"
              matline=""
            >
              
              
            </div>
            <mat-divider
              aria-orientation="horizontal"
              class="mat-divider mat-divider-horizontal"
              role="separator"
            />
          </div>
        </div>
      </mat-list-option>
    </mat-selection-list>
  </div>
</div>
`;

exports[`VersionsInfoComponent restoring a version should show restore button for other versions when selected 1`] = `
<div
  id="root5"
>
  <div
    class="mat-typography"
  >
    
    
    <mat-selection-list
      aria-disabled="false"
      aria-multiselectable="false"
      class="qa-version-list mat-selection-list mat-list-base"
      color="primary"
      role="listbox"
      tabindex="0"
    >
      <h3
        class="qa-version-list-header mat-subheader"
        matsubheader=""
      >
         VERSIONS_LIST_HEADER 
      </h3>
      <mat-divider
        aria-orientation="horizontal"
        class="mat-divider mat-divider-horizontal"
        role="separator"
      />
      
      <mat-list-option
        aria-disabled="false"
        aria-selected="false"
        class="qa-version-testId mat-primary"
        role="option"
        tabindex="-1"
      >
        <div
          class="mat-list-item-content mat-list-item-content-reverse"
        >
          <div
            class="mat-list-item-ripple mat-ripple"
            mat-ripple=""
          />
          
          <div
            class="mat-list-text"
          >
            <p
              class="qa-version-timestamp mat-line"
              matline=""
            >
               Nov 8, 2020 04:03 
            </p>
            <p
              class="qa-version-username mat-line"
              matline=""
            >
               testUserName 
            </p>
            <div
              class="version-actions mat-line"
              matline=""
            >
              
              
            </div>
            <mat-divider
              aria-orientation="horizontal"
              class="mat-divider mat-divider-horizontal"
              role="separator"
            />
          </div>
        </div>
      </mat-list-option>
      <mat-list-option
        aria-disabled="false"
        aria-selected="true"
        class="qa-version-testVariantId mat-primary mat-list-single-selected-option"
        role="option"
        tabindex="-1"
      >
        <div
          class="mat-list-item-content mat-list-item-content-reverse"
        >
          <div
            class="mat-list-item-ripple mat-ripple"
            mat-ripple=""
          />
          
          <div
            class="mat-list-text"
          >
            <p
              class="qa-version-timestamp mat-line"
              matline=""
            >
               Nov 8, 2020 04:03 
            </p>
            <p
              class="qa-version-username mat-line"
              matline=""
            >
               testUserName2 
            </p>
            <div
              class="version-actions mat-line"
              matline=""
            >
              
              
              <button
                class="qa-restore-version-action"
                mat-stroked-button=""
              >
                 RESTORE 
              </button>
            </div>
            <mat-divider
              aria-orientation="horizontal"
              class="mat-divider mat-divider-horizontal"
              role="separator"
            />
          </div>
        </div>
      </mat-list-option>
    </mat-selection-list>
  </div>
</div>
`;

exports[`VersionsInfoComponent selecting version should show the selected version 1`] = `
<mat-selection-list
  aria-disabled="false"
  aria-multiselectable="false"
  class="qa-version-list mat-selection-list mat-list-base"
  color="primary"
  role="listbox"
  tabindex="0"
>
  <h3
    class="qa-version-list-header mat-subheader"
    matsubheader=""
  >
     VERSIONS_LIST_HEADER 
  </h3>
  <mat-divider
    aria-orientation="horizontal"
    class="mat-divider mat-divider-horizontal"
    role="separator"
  />
  
  <mat-list-option
    aria-disabled="false"
    aria-selected="false"
    class="qa-version-testId mat-primary"
    role="option"
    tabindex="-1"
  >
    <div
      class="mat-list-item-content mat-list-item-content-reverse"
    >
      <div
        class="mat-list-item-ripple mat-ripple"
        mat-ripple=""
      />
      
      <div
        class="mat-list-text"
      >
        <p
          class="qa-version-timestamp mat-line"
          matline=""
        >
           Nov 8, 2020 04:03 
        </p>
        <p
          class="qa-version-username mat-line"
          matline=""
        >
           testUserName 
        </p>
        <div
          class="version-actions mat-line"
          matline=""
        >
          
          
        </div>
        <mat-divider
          aria-orientation="horizontal"
          class="mat-divider mat-divider-horizontal"
          role="separator"
        />
      </div>
    </div>
  </mat-list-option>
  <mat-list-option
    aria-disabled="false"
    aria-selected="true"
    class="qa-version-testVariantId mat-primary mat-list-single-selected-option"
    role="option"
    tabindex="-1"
  >
    <div
      class="mat-list-item-content mat-list-item-content-reverse"
    >
      <div
        class="mat-list-item-ripple mat-ripple"
        mat-ripple=""
      />
      
      <div
        class="mat-list-text"
      >
        <p
          class="qa-version-timestamp mat-line"
          matline=""
        >
           Nov 8, 2020 04:03 
        </p>
        <p
          class="qa-version-username mat-line"
          matline=""
        >
           testUserName2 
        </p>
        <div
          class="version-actions mat-line"
          matline=""
        >
          
          
          <button
            class="qa-restore-version-action"
            mat-stroked-button=""
          >
             RESTORE 
          </button>
        </div>
        <mat-divider
          aria-orientation="horizontal"
          class="mat-divider mat-divider-horizontal"
          role="separator"
        />
      </div>
    </div>
  </mat-list-option>
</mat-selection-list>
`;

exports[`VersionsInfoComponent showing versions should show header indicating no versions are available 1`] = `
<div
  id="root0"
>
  <div
    class="mat-typography"
  >
    
    <mat-selection-list
      aria-disabled="false"
      aria-multiselectable="true"
      class="qa-version-list mat-selection-list mat-list-base"
      role="listbox"
      tabindex="-1"
    >
      <h3
        class="qa-version-list-header mat-subheader"
        matsubheader=""
      >
         NO_VERSIONS_AVAILABLE 
      </h3>
      <mat-divider
        aria-orientation="horizontal"
        class="mat-divider mat-divider-horizontal"
        role="separator"
      />
    </mat-selection-list>
    
  </div>
</div>
`;

exports[`VersionsInfoComponent showing versions should show list of versions 1`] = `
<div
  id="root1"
>
  <div
    class="mat-typography"
  >
    
    
    <mat-selection-list
      aria-disabled="false"
      aria-multiselectable="false"
      class="qa-version-list mat-selection-list mat-list-base"
      color="primary"
      role="listbox"
      tabindex="0"
    >
      <h3
        class="qa-version-list-header mat-subheader"
        matsubheader=""
      >
         VERSIONS_LIST_HEADER 
      </h3>
      <mat-divider
        aria-orientation="horizontal"
        class="mat-divider mat-divider-horizontal"
        role="separator"
      />
      
      <mat-list-option
        aria-disabled="false"
        aria-selected="true"
        class="qa-version-testId mat-primary mat-list-single-selected-option"
        role="option"
        tabindex="-1"
      >
        <div
          class="mat-list-item-content mat-list-item-content-reverse"
        >
          <div
            class="mat-list-item-ripple mat-ripple"
            mat-ripple=""
          />
          
          <div
            class="mat-list-text"
          >
            <p
              class="qa-version-timestamp mat-line"
              matline=""
            >
               Nov 8, 2020 04:03 
            </p>
            <p
              class="qa-version-username mat-line"
              matline=""
            >
               testUserName 
            </p>
            <div
              class="version-actions mat-line"
              matline=""
            >
              
              <button
                class="qa-new-version-action"
                mat-stroked-button=""
              >
                 VERSION 
              </button>
              
            </div>
            <mat-divider
              aria-orientation="horizontal"
              class="mat-divider mat-divider-horizontal"
              role="separator"
            />
          </div>
        </div>
      </mat-list-option>
      <mat-list-option
        aria-disabled="false"
        aria-selected="false"
        class="qa-version-testVariantId mat-primary"
        role="option"
        tabindex="-1"
      >
        <div
          class="mat-list-item-content mat-list-item-content-reverse"
        >
          <div
            class="mat-list-item-ripple mat-ripple"
            mat-ripple=""
          />
          
          <div
            class="mat-list-text"
          >
            <p
              class="qa-version-timestamp mat-line"
              matline=""
            >
               Nov 8, 2020 04:03 
            </p>
            <p
              class="qa-version-username mat-line"
              matline=""
            >
               testUserName2 
            </p>
            <div
              class="version-actions mat-line"
              matline=""
            >
              
              
            </div>
            <mat-divider
              aria-orientation="horizontal"
              class="mat-divider mat-divider-horizontal"
              role="separator"
            />
          </div>
        </div>
      </mat-list-option>
    </mat-selection-list>
  </div>
</div>
`;
